automation/taskcluster/lib/tasks.py

```diff
@@ -477,9 +482,16 @@ def craft_raptor_signing_task(
                 'machine': {
                     'platform': 'android-all',
                 },
+                'collection': {
```

Comment:
That's a way to display optimized builds from debug ones on Treeherder. Treeherder doesn't require it, but taskgraph does.

Reply:
Oh, does this color/change the font/group optimized builds on Treeherder? Also, if it's a treeherder option, I don't understand what you mean by "Treeherder doesn't require it, but taskgraph does". :thinking:

Reply:
It just change the platform label, by adding an `opt` at the end of the name.  Although, I double checked and it turns out, in this specific case, nothing has changed for the raptor jobs:
* https://treeherder.mozilla.org/#/jobs?repo=fenix&revision=acfd73419c022bd26bf4e7473dc2fd397446b3b6 => `opt` was already there in "Android 7.0 MotoG5 opt"
* https://treeherder.mozilla.org/#/jobs?repo=fenix&revision=4462899763039336645530cec59414b1f22952e3 => the label hasn't changed.

taskgraph/treeherder is a great question! So taskcluster accepts anything in extra https://github.com/taskcluster/taskcluster/blob/6d7ab8182f4a637fb9e66df7070227bde285f408/services/queue/schemas/v1/task.yml#L159-L171. Taskcluster has a service to submit `extra.treeherder` to Treeherder (https://github.com/taskcluster/taskcluster-treeherder - can't where it was migrated to). Treeherder doesn't mind accepting an `extra.treeherder` which doesn't have a collection set. It seems it defaults to opt.

That said, taskgraph is pretty pedantic about the tasks it submits to taskcluster (and thus to treeherder) and forces some keys (like `extra.treeherder.collection`) to be set. So that's how "Treeherder doesn't require it, but taskgraph does".

Feel free to ask any questions if my explanation isn't clear.